1.0: Add OpenSearch plugin documentation

cosmo0920 opened the pull request:
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

cosmo0920 committed:
1.0: Add OpenSearch plugin documentation

Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

cosmo0920 commented on output/opensearch.md:
```diff
@@ -0,0 +1,163 @@
+
+## Installation
+
+Since `out_opensearch` has been included in the standard distribution of `td-agent` since v4.4.0 and the alternative distribution of `calyptia-fluentd` since v1.3.4, `td-agent`/`calyptia-fluentd` users do not need to install it manually.
```
But, I'm not sure the exact version of OpenSearch plugin included version of td-agent.

agup006 replied:
It would be good to merge prior to being available in td-agent, and then adding once available

cosmo0920 replied:
Sure. I fixed it in https://github.com/fluent/fluentd-docs-gitbook/pull/390/commits/76ef4607b1fe4620d8b4b12e82553109746f51f6.

ashie replied:
FYI:
Probably the next version of td-agent (which includes opensearch plugin) will be v4.3.1 but not determined yet.

The minor version of td-agent is tied with the one of fluentd.
When the minor version of fluentd is incremeneted, td-agent's one is also incremented.

kenhys commented on output/opensearch.md:
```diff
@@ -0,0 +1,163 @@
+# opensearch
+
+The `out_opensearch` Output plugin writes records into Opensearch. By default, it creates records using [bulk api](https://opensearch.org/docs/latest/opensearch/rest-api/document-apis/bulk/) which performs multiple indexing operations in a single API call. This reduces overhead and can greatly increase indexing speed. This means that when you first import records using the plugin, records are not ...
```
```suggestion
The `out_opensearch` Output plugin writes records into OpenSearch. By default, it creates records using [bulk api](https://opensearch.org/docs/latest/opensearch/rest-api/document-apis/bulk/) which performs multiple indexing operations in a single API call. This reduces overhead and can greatly increase indexing speed. This means that when you first import records using the plugin, records are not immediately pushed to OpenSearch.
```

kenhys commented on output/opensearch.md:
```diff
@@ -0,0 +1,163 @@
+## Example Configuration
+
+Here is a simple working configuration which should serve as a good starting point for most users:
+
```
It may be obvious, but it is kind to note "for OpenSearch instance on localhost" or something.

cosmo0920 replied:
Fixed in https://github.com/fluent/fluentd-docs-gitbook/pull/390/commits/798582a31dffd7c5dc2a71ccc267e4ac37ced8e2.

kenhys reviewed (commented):
I've added a few comments.

cosmo0920 committed:
Use "OpenSearch" notation

Co-authored-by: Kentaro Hayashi <kenhys@gmail.com>
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

cosmo0920 committed:
Use more concrete sentence for "the first configuration"

Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

cosmo0920 committed:
Update opensearch plugin installation step

Current stable version of td-agent does not include OpenSearch plugin
yet. We should add / modify descriptions for td-agent users later.

Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

cosmo0920 commented:
@kenhys @ashie Can we merge this OpenSearch plugin descriptions? It seems that OpenSearch plugin gets initial users increasing momentum. We should follow this momentum and want to merge this doc into official docs soon.

cosmo0920 commented:
Thanks!
